src/cloud-cli/meltano/cloud/cli/history/command.py

```diff
@@ -136,21 +157,32 @@ async def history(
     schedule: str | None,
     schedule_prefix: str | None,
     schedule_contains: str | None,
+    deployment: str | None,
+    deployment_prefix: str | None,
+    deployment_contains: str | None,
+    result: str | None,
     limit: int,
     output_format: str,
     lookback: datetime.timedelta | None,
 ) -> None:
     """Get a Meltano project execution history in Meltano Cloud."""
-    schedule_filter = _schedule_arg_to_glob(
+    schedule_filter = _arg_to_glob_filter(
         schedule,
         schedule_prefix,
         schedule_contains,
     )
+    deployment_filter = _arg_to_glob_filter(
+        deployment,
+        deployment_prefix,
+        deployment_contains,
+    )
 
     now = datetime.datetime.now(tz=utils.UTC)
     items = await HistoryClient.get_history_list(
         context.config,
         schedule_filter=schedule_filter,
+        environment_filter=deployment_filter,
```

Comment:
This feels like a dangerous equality. At some point users might want to actually filter on the Meltano environment name across deployments. What would happen then? (I realize this is theoretical right now but a deployment is not 1 to 1 with an environment.

Reply:
Makes sense, I've started https://github.com/meltano/infra/pull/845 to actually be able to filter by deployment